### ISSUE TITLE
Change to build to use dotnet

### ISSUE THREAD

**Issue by victorhurdugaci** [aspnet/Performance#17]
At a minimum, you have to upgrade a few scripts and project.json files just like here: aspnet/Configuration@9cfd22f

cc @NTaylorMullen

**Comment by SajayAntony** [comment]
@JunTaoLuo Does this address cross plat and should I reopen this issue?
@muratg @DavidObando 
```
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.netcore.platforms/index.json 1741ms
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.netcore.platforms/1.0.1-rc2-23805/microsoft.netcore.platforms.1.0.1-rc2-23805.nupkg.
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/index.json 1846ms
info :   CACHE https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/index.json
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg.
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/index.json 1923ms
info :   CACHE https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/index.json
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg.
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.netcore.platforms/index.json 1927ms
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/index.json 2098ms
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/index.json 1975ms
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/index.json 2058ms
info :   CACHE https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/index.json
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg.
info :   OK https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/index.json 2157ms
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.netcore.platforms/1.0.1-rc2-23805/microsoft.netcore.platforms.1.0.1-rc2-23805.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.netcore.platforms/1.0.1-rc2-23805/microsoft.netcore.platforms.1.0.1-rc2-23805.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.netcore.platforms/1.0.1-rc2-23805/microsoft.netcore.platforms.1.0.1-rc2-23805.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.netcore.platforms/1.0.1-rc2-23805/microsoft.netcore.platforms.1.0.1-rc2-23805.nupkg.
error: Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg'.
An error occurred while sending the request.
error: Unable to load package 'Microsoft.AspNetCore.Server.Kestrel'.
sajay@sajay-nix:~/code/aspnet/performance/testapp/HelloWorldMvc$ sajay@sajay-nix:~/code/aspnet/performance/testapp/HelloWorldMvc$ dotnet restore
-bash: sajay@sajay-nix:~/code/aspnet/performance/testapp/HelloWorldMvc$: No such file or directory
sajay@sajay-nix:~/code/aspnet/performance/testapp/HelloWorldMvc$ log  : Restoring packages for /home/sajay/code/aspnet/performance/testapp/HelloWorldMvc/project.json...
No command 'log' found, but there are 16 similar ones
log: command not found
sajay@sajay-nix:~/code/aspnet/performance/testapp/HelloWorldMvc$ info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/index.json.
```

**Comment by JunTaoLuo** [comment]
@SajayAntony That looks like myget issues, I have not been able to reproduce this locally. I believe there are some issues with packages on the current feeds @pranavkm?

**Comment by SajayAntony** [comment]
@pranavkm @victorhurdugaci. @JunTaoLuo @muratg I'm reopening this since the cross plat is currently broken, we verified this on Pranav's machine yesterday.

**Comment by SajayAntony** [comment]
@DavidObando @pranavkm we need to ensure that https://github.com/aspnet/Performance/tree/dev/testapp/HelloWorldMvc compiles cross plat

**Comment by victorhurdugaci** [comment]
Is it broken on Linux or Mac? Or both?

**Comment by SajayAntony** [comment]
I tried only on ubuntu 14.04. I haven't tried it on the mac yet.

**Comment by victorhurdugaci** [comment]
@pranavkm is the download broken? Or the feed? Or what? From the log it looks like missing packages or myget issues

**Comment by NTaylorMullen** [comment]
I was able to reproduce the failure on travis and on a local VM. Feed seems fine, local windows builds passing. Reprod it with the Razor repo. Example: https://travis-ci.org/aspnet/Razor/jobs/108303710

This seems like an issue with one of dotnetcli/nuget/coreclr.

**Comment by SajayAntony** [comment]
So is there a temporary work around for this?

**Comment by NTaylorMullen** [comment]
@SajayAntony If you're able to upgrade your ubuntu environment to 15+ then you should be ok.

**Comment by SajayAntony** [comment]
Wait isn't cli broken on 15? https://github.com/dotnet/cli/issues/308
Is there a specific version of the runtime+Coreclr that we can use on 14?

**Comment by NTaylorMullen** [comment]
@pakrym was abel to run a dotnet build/test suite on some version of 15, he can fill in the gaps.

**Comment by muratg** [comment]
@NTaylorMullen would be pinging folks, current thinking is there's a CoreFX issue on that specific Ubuntu build.

**Comment by DavidObando** [comment]
I noticed that if I run sh build.sh multiple times, it fails in different packages. The behavior is inconsistent.

**Comment by muratg** [comment]
@NTaylorMullen Do we know if the new nuget works better? Was CLI able to consume it?

**Comment by NTaylorMullen** [comment]
@muratg I don't believe they'ved pulled a new NuGEt yet. My understanding was that they're waiting for https://github.com/dotnet/cli/pull/1332 to go in.

**Comment by DavidObando** [comment]
I also found that `dotnet restore -s https://myget.org/f/dotnet-core` produces a predictable behavior. With this source I'm able to generate a lock file, but the restore also fails to resolve a number of dependencies
```
Errors in /home/davidobando/github/aspnet/Performance/testapp/HelloWorldMvc/project.json
    Unable to resolve Microsoft.AspNetCore.Server.Kestrel (>= 1.0.0) for DNXCore, Version=v5.0.
    Unable to resolve Microsoft.AspNetCore.Mvc (>= 1.0.0) for DNXCore, Version=v5.0.
    Unable to resolve Microsoft.AspNetCore.IISPlatformHandler (>= 1.0.0) for DNXCore, Version=v5.0.
    Unable to resolve Microsoft.AspNetCore.Server.Kestrel (>= 1.0.0) for DNX, Version=v4.5.1.
    Unable to resolve Microsoft.AspNetCore.Mvc (>= 1.0.0) for DNX, Version=v4.5.1.
    Unable to resolve Microsoft.AspNetCore.IISPlatformHandler (>= 1.0.0) for DNX, Version=v4.5.1.
```
So I added that source to NuGet.config and noticed the same error as before, but this time around only in those three packages listed above:
```
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.mvc/1.0.0-rc2-17051/microsoft.aspnetcore.mvc.1.0.0-rc2-17051.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg.
log  : Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg'.
An error occurred while sending the request.
info :   GET https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.iisplatformhandler/1.0.0-rc2-16217/microsoft.aspnetcore.iisplatformhandler.1.0.0-rc2-16217.nupkg.
error: Failed to download package from 'https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg'.
An error occurred while sending the request.
error: Unable to load package 'Microsoft.AspNetCore.Server.Kestrel'.
```
I decided to `curl` it and this is what I see:
```
davidobando@davidobando-ubuntu-nge:~$ curl -sSl https://www.myget.org/F/aspnetcidev/api/v3/flatcontainer/microsoft.aspnetcore.server.kestrel/1.0.0-rc2-16391/microsoft.aspnetcore.server.kestrel.1.0.0-rc2-16391.nupkg
<html><head><title>Object moved</title></head><body>
<h2>Object moved to <a href="https://myget-2e16.kxcdn.com/privatefeedaspnetcidev/Microsoft.AspNetCore.Server.Kestrel-1.0.0-rc2-16391?sv=2014-02-14&amp;sr=b&amp;sig=nSV0kkYt5LKqWuum7IkGTNHI9aw82%2Fslx%2BSN5V9wl7o%3D&amp;st=2016-02-11T19%3A55%3A00Z&amp;se=2016-02-11T21%3A05%3A00Z&amp;sp=r&amp;rscc=max-age%3D3600&amp;rsct=binary%2Foctet-stream&amp;rscd=filename%3DMicrosoft.AspNetCore.Server.Kestrel.1.0.0-rc2-16391.nupkg">here</a>.</h2>
</body></html>
```
Essentially, myget is redirecting us to a CND node with an HTTP status code of 302 (according to Fiddler). The 302 appears to be working in the windows CLI but not in the Linux CLI.

@NTaylorMullen do you know if there's a bug for this in CLI?

**Comment by DavidObando** [comment]
Added dotnet/cli#1372 in case it wasn't logged before. I couldn't find anything like it.

**Comment by NTaylorMullen** [comment]
@DavidObando if you do -s to the private feed (no redirect) is all well?

**Comment by DavidObando** [comment]
@NTaylorMullen yes, I tried -s'ing to the repos, same effect as having them in NuGet.config.

**Comment by yishaigalatzer** [comment]
The code is exactly the same for windows and Linux so I'm guessing there is an issue with http client because none of the cider is handling redirects directly. We need to involve the owners of http client.

We can probably build a simple repro in a console app.

**Comment by SajayAntony** [comment]
@yishaigalatzer I'm including @davidsh. 
This is currently blocking our perf/stress on linux at this point.

**Comment by yishaigalatzer** [comment]
Thanks @sajayantony

**Comment by davidsh** [comment]
Please open up a separate issue in the dotnet/corefx repo for HttpClient issues.  Since it x-plat (Linux) we will assign it to the x-plat team.  Can you provide a small repro to help diagnose? Thx.

**Comment by yishaigalatzer** [comment]
@emgarten can you help track the repro we had before for this scenario?

**Comment by NTaylorMullen** [comment]
@DavidObando since you were the one to discover the underlying issue would you mind filing an issue with your repo over at https://github.com/dotnet/corefx/issues?

**Comment by DavidObando** [comment]
@NTaylorMullen will do.

**Comment by DavidObando** [comment]
dotnet/corefx#6079

**Comment by DavidObando** [comment]
This issue is caused by System.Net.Http.Native.so being linked against the wrong libcurl in builds up to 23808. Build 23809 of runtime.ubuntu.14.04-x64.runtime.native.System.Net.Http is the first build where things work as expected. I'm now working to see if upgrading the CLI to a build that uses the fixed System.Net.Http fixes `dotnet restore`.

**Comment by DavidObando** [comment]
I confirmed that moving KoreBuild to pull a version of dotnet that contains fixed bits unblocks the `dotnet restore` issue. Added aspnet/Universe#384 to request an update to KoreBuild.sh.

**Comment by emgarten** [comment]
@yishaigalatzer the network issues should be fixed once the latest version of nuget goes into the CLI.

**Comment by DavidObando** [comment]
@emgarten actually nuget isn't involved. It's just System.Net.Http. Latest CLI works well.

**Comment by JunTaoLuo** [comment]
Completed in #26
